File: C++/Arrays/Problem20.cpp


// Buy and sell Stock


  #include <iostream>
  #include <bits/stdc++.h>
  using namespace std;
  
  int main() 
  {
      int n=6;
      int arr[n]={7,1,5,3,6,4};
       
      int maxProfit=0;
      int minPrice=INT_MAX;
      for(int i=0;i<n;i++){
        minPrice=min(minPrice,arr[i]);
        maxProfit=max(maxProfit,arr[i]-minPrice);
      }
      std::cout << maxProfit << std::endl;
      return 0;
  }

File: Javascript/Arrays/Problem20.js
// buy and sell the stocks max profit 


let arr=[7,1,5,3,6,4];

// ans =5;

let maxPro=0;
let min=Infinity;

for(let i=0;i<arr.length;i++){
  min=Math.min(min,arr[i]);
  let val=arr[i]-min;
  maxPro=Math.max(maxPro,val);
}


console.log(maxPro)
